src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -833,6 +833,46 @@ void ObjectCopyRequest<I>::compute_dst_object_may_exist() {
                    << dendl;
 }
 
+template <typename I>
+void ObjectCopyRequest<I>::trigger_copyup() {
+  // With mirroring, a write to a non-existent primary cloned image object will 
+  // trigger a copyup which updates the older synced snaps with the parent data.
+  // This causes the rbd-mirror to ignore the parent data and only sync the new write
+  // to the clone, causing data corruption.
+  // For rbd-mirror scenarios, we trigger a copyup on the object and write the parent
+  // data to the object before writing the new clone image data.
+ldout(m_cct, 20) << "NITHYA: m_src_snap_id_start: " << m_src_snap_id_start << ", parent:" << m_src_image_ctx->parent << dendl;
+  if (m_src_snap_id_start != 0 && m_src_image_ctx->parent != nullptr) {
+    ldout(m_cct, 20) << dendl;
+    auto io_context = m_dst_image_ctx->duplicate_data_io_context();
+    auto ctx = create_context_callback<
+      ObjectCopyRequest<I>, &ObjectCopyRequest<I>::handle_trigger_copyup>(this);
+    bool r = io::util::trigger_copyup(
+      m_dst_image_ctx, m_dst_object_number, io_context, ctx);
+    if (r == false) {
+      // No parent found
+      send_update_object_map();
```

Comment:
nit: I would add `return` here for clarity.

Reply:
Done.